kanopi/wp-vanilla-theme
# Implement Kanopi Pack

## Description
> As a developer, I want an updated asset bundling system.

This PR contains Theme level changes for the base WordPress theme to implement Kanopi Pack. The project Readme is updated to reflect the new configuration options and defaults. 

A few choices I made:
1. Proxy based configuration is documented in comments, this is specific to the Docksal implementation
2. Theme namespaces were excluded to limit the number/scope of changes in the PR. The theme will not run the in theme auto-loader if the Asset Loader is already loaded (i.e. from site/root Composer).
3. The Asset Loader is currently a public repository under my Github user. For CBInsights, I will implement a feature to read the Webpack manifest for dynamic, cache-busted file names. When I implement that, I will fork and rename over to Kanopi.

## Testing

1. Setup a test WordPress site locally, pull this theme, then activate it for the site.
2. In `wp-config.php`, add the following constant: `define('KANOPI_DEVELOPMENT_ASSET_URL', `'http://localhost:4400');`
2. Switch to the theme directory
3. Run `npm ci` to install all packages. Make sure this is run form a machine with SSH access to the Kanopi organization
4. Run `npm run development`, in the CLI, you should see the assets **Compiled successfully** and continue running
5. Load the site and confirm assets are loaded from `localhost:4400`. Additionally, the Console log will show a message stating **[WDS] Live Reloading enabled.**
6. Stop the WDS server (Ctrl+C) and run `npm run production`
7. Comment/remove the `KANOPI_DEVELOPMENT_ASSET_URL` constant.
8. Visit the site again and confirm the assets are loaded from the site `assets/dist` folder directly.
